### PR TITLE
[#SPARK-16911] Fix the links in the programming guide

### PR DESCRIPTION
**Pull request by shiv4nsh**
## What changes were proposed in this pull request?

 Fix the broken links in the programming guide of the Graphx Migration and understanding closures
## How was this patch tested?

By running the test cases  and checking the links.

### CODE REVIEW

File: docs/programming-guide.md [hunk 1]
```diff
@@ -1097,7 +1097,7 @@ for details.
 <tr>
   <td> <b>foreach</b>(<i>func</i>) </td>
   <td> Run a function <i>func</i> on each element of the dataset. This is usually done for side effects such as updating an <a href="#accumulators">Accumulator</a> or interacting with external storage systems.
-  <br /><b>Note</b>: modifying variables other than Accumulators outside of the <code>foreach()</code> may result in undefined behavior. See <a href="#ClosuresLink">Understanding closures </a> for more details.</td>
+  <br /><b>Note</b>: modifying variables other than Accumulators outside of the <code>foreach()</code> may result in undefined behavior. See <a href="#understanding-closures-a-nameclosureslinka">Understanding closures </a> for more details.</td>
```
Comment (srowen) [inline docs/programming-guide.md]: These anchors don't exist, are you sure about this? or close this PR please

Comment (shiv4nsh) [inline docs/programming-guide.md]: @srowen : When we see on the documentation: https://spark.apache.org/docs/latest/programming-guide.html#understanding-closures-a-nameclosureslinka Points to the understanding Closure documentation part hence I replaced it here. I have also rechecked it by generating the docs

Comment (srowen) [inline docs/programming-guide.md]: Ah right this is an auto-generated anchor. this is OK.

File: docs/programming-guide.md [hunk 2]
```diff
@@ -1571,7 +1571,7 @@ have changed from returning (key, list of values) pairs to (key, iterable of val
 </div>
 
 Migration guides are also available for [Spark Streaming](streaming-programming-guide.html#migration-guide-from-091-or-below-to-1x),
-[MLlib](ml-guide.html#migration-guide) and [GraphX](graphx-programming-guide.html#migrating-from-spark-091).
+[MLlib](ml-guide.html#migration-guide) and [GraphX](graphx-programming-guide.html#migrating-from-spark-11).
```
Comment (srowen) [inline docs/programming-guide.md]: This is also OK, though at this stage, these links and the entire migration sections they refer to are very old. We can just remove them.

**Comment by shiv4nsh** [comment]
@srowen : So can we merge this ? or do I have to make some  more changes?

**Comment by srowen** [comment]
Did you see my last comment? I think we should remove the old migration guides in latest Spark 2 docs.

**Comment by shiv4nsh** [comment]
@srowen : So do we have to remove this whole part?
screenshot from 2016-08-05 13-04-47

**Comment by srowen** [comment]
Personally, I think that any documentation that talks about migrating _to_ a Spark 1.x version can be removed from the 2.x documentation in master. The target of such a migration is also obsolete. You could make this JIRA really about that, as well as fixing the first link as a side effect.

**Comment by shiv4nsh** [comment]
@srowen : I have updated the documentation and here is what it looks like : 
screenshot from 2016-08-05 14-21-36

Please review it ! Thanks :)

**Comment by srowen** [comment]
Yes, what about removing the streaming/graphx migration guides? they're also about very old versions.

**Comment by shiv4nsh** [comment]
@srowen : I have removed them too and now it looks like this:

**Graphx Guide :**

graphx

**Streaming Guide:** 

spark_streaming

Please review them ! Thanks :)

**Comment by srowen** [comment]
Jenkins test this please

**Comment by shiv4nsh** [comment]
@srowen : Can we merge this now ?

**Comment by srowen** [comment]
Yes, though we conventionally leave things open for review for a time. You don't need to ping so quickly.

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**